examples/vision/sam.py

```diff
@@ -0,0 +1,595 @@
+            inputs["ground_truth_mask"] = ground_truth_mask
+
+            yield inputs
+
+    def get_bounding_box(self, ground_truth_map):
+        # get bounding box from mask
+        y_indices, x_indices = np.where(ground_truth_map > 0)
+        x_min, x_max = np.min(x_indices), np.max(x_indices)
+        y_min, y_max = np.min(y_indices), np.max(y_indices)
+
+        # add perturbation to bounding box coordinates
+        H, W = ground_truth_map.shape
+        x_min = max(0, x_min - np.random.randint(0, 20))
+        x_max = min(W, x_max + np.random.randint(0, 20))
+        y_min = max(0, y_min - np.random.randint(0, 20))
+        y_max = min(H, y_max + np.random.randint(0, 20))
+        bbox = [x_min, y_min, x_max, y_max]
+
+        return bbox
+
+
+"""
+`get_bounding_box()` is responsible for turning the ground-truth segmentation maps into
+bounding boxes. These bounding boxes are fed to SAM as prompts (along with the original
+images) during fine-tuning and SAM is then trained to predict valid masks.
+
+The advantage of this first creating a generator and then using it to create a
+`tf.data.Dataset` is the flexbility. Sometimes, we may need to use utitlities from other
+libraries ([`albumentations`](https://albumentations.ai/), for example) which may not
+come in native TensorFlow implementations. By using this workflow, we can easily
+accommodate such use case.
+
+But the non-TF counterparts might introduce performance bottlenecks, though. However, for
+our example, it should work just fine.
+
+Now, we prepare the `tf.data.Dataset` from our training set.
+"""
+
+# Define the output signature of the generator class.
+output_signature = {
+    "pixel_values": tf.TensorSpec(shape=(3, None, None), dtype=tf.float32),
+    "original_sizes": tf.TensorSpec(shape=(None,), dtype=tf.int64),
+    "reshaped_input_sizes": tf.TensorSpec(shape=(None,), dtype=tf.int64),
+    "input_boxes": tf.TensorSpec(shape=(None, None), dtype=tf.float64),
+    "ground_truth_mask": tf.TensorSpec(shape=(None, None), dtype=tf.int32),
+}
+
+# Prepare the dataset object.
+train_dataset_gen = Generator(dataset_path, processor)
+train_ds = tf.data.Dataset.from_generator(
+    train_dataset_gen, output_signature=output_signature
+)
+
+"""
+Next, we configure the dataset for performance.
+"""
+
+auto = tf.data.AUTOTUNE
+batch_size = 2
+shuffle_buffer = 4
+
+train_ds = (
+    train_ds.cache()
+    .shuffle(shuffle_buffer)
+    .batch(batch_size)
+    .prefetch(buffer_size=auto)
+)
+
+"""
+Take a single batch of data and inspect the shapes of the elements present inside of it.
+"""
+
+sample = next(iter(train_ds))
+for k in sample:
+    print(k, sample[k].shape, sample[k].dtype, isinstance(sample[k], tf.Tensor))
+
+"""
+### Training
+"""
+
+"""
+We will now write DICE loss. This implementation is based on [MONAI DICE
+loss](https://docs.monai.io/en/stable/_modules/monai/losses/dice.html#DiceLoss).
+"""
+
+from tensorflow.keras import backend as K
```

Comment:
Don't use this namespace, it's deprecated. Instead use tf ops like `tf.reduce_sum`